File: app/src/main/java/com/heinika/designpattern/factory/FactoryPatternDemo.java
package com.heinika.designpattern.factory;

public class FactoryPatternDemo {
    public void main(){
        Circle circle = (Circle) ShapeFactory.getshape("Circle");
        Square square = (Square) ShapeFactory.getshape("Square");
        Rectangle rectangle = (Rectangle) ShapeFactory.getshape("Rectangle");
        circle.draw();
        square.draw();
        rectangle.draw();
    }
}

File: app/src/main/java/com/heinika/designpattern/kotlin/factory/FactoryPatternDemo.kt
package com.heinika.designpattern.kotlin.factory

fun main(args: Array<String>){
    val square = ShapeFactory().getShape("Square")
    val circle = ShapeFactory().getShape("Circle")
    val rectangle = ShapeFactory().getShape("Rectangle")
    square!!.draw()
    circle!!.draw()
    rectangle!!.draw()
}

class FactoryPatternDemo{}